src/Statistics/Histogram.cpp

```diff
@@ -54,4 +56,19 @@ static Histogram BuildHistogramWithNumberOfBins(const std::optional<DataSet>& da
   return best_histogram;
 }
 
+[[nodiscard]] std::vector<int> GetBinWidth(size_t number_of_bins, int histogram_width) {
+  const int narrower_width = histogram_width / number_of_bins;
+  const int wider_width = narrower_width + 1;
+
+  const int number_of_wider_bins = histogram_width % number_of_bins;
+  const int number_of_narrower_bins = number_of_bins - number_of_wider_bins;
+
+  std::vector<int> result(number_of_narrower_bins, narrower_width);
+  const std::vector<int> wider_widths(number_of_wider_bins, wider_width);
+  result.insert(std::end(result), std::begin(wider_widths), std::end(wider_widths));
+
+  std::shuffle(std::begin(result), std::end(result), std::default_random_engine{});
```

Comment:
Nit: To get the same behaviour on all platforms I would prefer to use a fixed random engine with a fixed seed, mt19937 for example. The default seed is implementation defined as far as I know.